AsyncDisplayKit/Details/Transactions/_ASAsyncTransaction.mm

```diff
@@ -159,32 +207,29 @@ - (NSString *)description
   Operation operation;
   operation._block = block;
   operation._group = this;
-  entry._operations.push_back(operation);
+  operation._priority = priority;
+  entry.pushOperation(operation);
   
   ++_pendingOperations; // enter group
   
-  NSUInteger maxThreads = [NSProcessInfo processInfo].activeProcessorCount;
-  if (maxThreads < 2) { // it is reasonable to have at least two working threads, also
-    maxThreads = 2;     // [_ASDisplayLayerTests testTransaction] requires at least two threads
-  }
+  NSUInteger maxThreads = [NSProcessInfo processInfo].activeProcessorCount * 2;
 
-#if 0
-  // Bit questionable - we can give main thread more CPU time during tracking;
-  if (maxThreads > 1 && [[NSRunLoop mainRunLoop].currentMode isEqualToString:UITrackingRunLoopMode])
+  // Bit questionable maybe - we can give main thread more CPU time during tracking;
```

Comment:
I love this idea.  I think it is the right thing to do, 100%.  We do NOT want to excessively hurt concurrency / overall CPU utilization even in this case, but lightening the load a touch will allow the scheduler to more consistently succeed at getting the main thread as much run time as it needs (obviously the thread is set to a much higher priority anyway, but your work has certainly confirmed the scheduler is imperfect when there are several threads at full load and main starts becoming active when it previously wasn't - or periodically needs to run every frame but confuses the scheduler because the overall load index looks a lot less compared to the long-lived background threads)